Backend/utils/db_utils.py

```diff
@@ -83,39 +82,36 @@ def preloadDataFromFile(fileName, table):
         else:
           message = "Unable to add item to table " + table.describe_table + ". Missing email, password, activationId, or active status"
           logger.error(message)
+      json_file.close()
   except IOError as e:
-    message = "Unable to load data into table"
-    logger.exception(message + '\nJSON:\n' + jsonData + '\n' + str(e))
-    return message
+    message = "Unable to open file"
+    logger.exception(message + ': ' + settings_file + '\n' + str(e))
+    return None
 
 def preloadDataFromJson(jsonData, table):
   if (type(jsonData) == str):
     items = json.loads(jsonData)
   else:
     items = jsonData
-  try:
-    for item in items:
-      if "email" in item.keys() and "password" in item.keys() and "activationId" in item.keys() and "active" in item.keys():
-        email = item['email']
-        password = item['password']
-        activationId = item['activationId']
-        active = item['active']
-
-        table.put_item(
-          Item={
-            'email': email,
-            'password': password,
-            'activationId': activationId,
-            'active': active
-          }
-        )
-      else:
-        message = "Unable to add item to table " + table.describe_table + ". Missing email, password, activationId, or active status"
-        logger.error(message)
-  except IOError as e:
-    message = "Unable to load data into table"
-    logger.exception(message + '\nJSON:\n' + jsonData + '\n' + str(e))
-    return message
+
+  for item in items:
+    if "email" in item.keys() and "password" in item.keys() and "activationId" in item.keys() and "active" in item.keys():
+      email = item['email']
+      password = item['password']
+      activationId = item['activationId']
+      active = item['active']
+
+      table.put_item(
+        Item={
+          'email': email,
+          'password': password,
+          'activationId': activationId,
+          'active': active
+        }
+      )
+    else:
+      message = "Unable to add item to table " + table.describe_table + ". Missing email, password, activationId, or active status"
+      logger.error(message)
 
 def getTable(tableName, dbInstance):
```

Comment:
ok so for stuff like this there isn't a need for two return statements. Functions should have as few paths as possible. So a better way to write this function would be as follows:
`
def getTable(tableName, dbInstance):
  var table = None

  try:
    blah blah blah
    table = dbInstance.Table(tableName)
  except whatever
    blah blah blah

  return table
`
This makes the function only have one entry point and one exit point. This may not seem super important but doing it makes the code much more resilent, easier to test and modify, and people like Vokolos happy because you aren't making unneeded Cyclomatic Complexity (https://en.wikipedia.org/wiki/Cyclomatic_complexity).

So this should happen for like every function around here refactor them so they only have the one entry and exit points.